### DISCUSSION TITLE
'Prerank' object has no attribute '_results'

### DISCUSSION THREAD

**Discussion by luorongxin-yuan** [zqfang/GSEApy#216] [Q&A] [answered]
Hi there,

I was able run the prerank function as below:

pre_res = gp.Prerank(rnk = ranking, gene_sets= ["CellMarker_Augmented_2021"], seed = 6, permutation_num=100)

then I was trying to unpack the prerank output into a pd dataframe:

out = []

for term in list(pre_res.results):
    out.append([terms, pre_res.results[term]["fdr"], pre_res.results[term]["es"], pre_res.results[term]["nes"]])

out_df = pd.DataFrame(out, columns=["term", "fdr", 'es', 'nes']).sort_values("fder").reset_index(drop = True)
out_df

However, it shows no attribute error:

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[50], line 3
      1 out = []
----> 3 for term in list(pre_res.results):
      4     out.append([terms, pre_res.results[term]["fdr"], pre_res.results[term]["es"], pre_res.results[term]["nes"]])
      6 out_df = pd.DataFrame(out, columns=["term", "fdr", 'es', 'nes']).sort_values("fder").reset_index(drop = True)

File ~/opt/anaconda3/envs/bioenv/lib/python3.11/site-packages/gseapy/base.py:574, in GSEAbase.results(self)
    569 @property
    570 def results(self):
    571     """
    572     compatible to old style
    573     """
--> 574     keys = list(self._results.keys())
    575     if len(keys) == 1:
    576         return self._results[keys[0]]

AttributeError: 'Prerank' object has no attribute '_results'

I was basically following a tutorial, so was very confused what was happening... the other attributes seem not to be available either.  Will greatly appreciate if someone knows what's going on.

**Comment by luorongxin-yuan** [comment, marked as answer]
sorry I'm stupid I capitalized the gp,prerank when it autofilled.... problem solved

**Reply by zqfang** [reply]
no worries.
